src/app/src/test/kotlin/org/imperial/mrc/hint/database/CalibrateDataRepositoryTests.kt

```diff
@@ -0,0 +1,55 @@
+package org.imperial.mrc.hint.database
+
+import org.springframework.test.context.ActiveProfiles
+import org.springframework.boot.test.context.SpringBootTest
+import org.springframework.transaction.annotation.Transactional
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.http.ResponseEntity
+import org.imperial.mrc.hint.db.CalibrateDataRepository
+import org.imperial.mrc.hint.exceptions.CalibrateDataException
+import org.imperial.mrc.hint.models.CalibrateResultRow
+import org.junit.jupiter.api.Test
+import org.jooq.tools.json.JSONObject
+import org.jooq.tools.json.JSONArray
+import org.assertj.core.api.Assertions.assertThat
+import org.assertj.core.api.Assertions.assertThatThrownBy
+import com.fasterxml.jackson.databind.ObjectMapper
+
+@ActiveProfiles(profiles = ["test"])
+@SpringBootTest
+class CalibrateDataRepositoryTests
+{
+    val expectedRow = JSONArray(listOf(CalibrateResultRow(
+        "indicator_test",
+        "calendar_quarter_test",
+        "age_group_test",
+        "sex_test",
+        "area_id_test",
+        0.2222,
+        0.5555,
+        0.1111,
+        0.9999
+    )))
+    val path = "/src/test/resources/duckdb/test.duckdb"
+
+    @Autowired
+    private lateinit var sut: CalibrateDataRepository
```

Comment:
I think you should mock class ?

Reply:
i am unsure as to why mock it vs this, i want to test this class right, what is the point of mocking it? sorry for all the questions btw! im still trying to properly learn spring!

Reply:
This style looks okay because we have no parameters but you could annotate with @Mock instead of @Autowired.

Reply:
i couldnt find the @mock annotation anywhere, but i have just initialised like `val sut = JooqCalibrateDataRepository()`!